crates/ruff_cli/tests/integration_test.rs

```diff
@@ -278,3 +278,108 @@ Found 1 error.
 
     Ok(())
 }
+
+#[test]
+fn different_safety_levels() -> Result<()> {
+    let mut cmd = Command::cargo_bin(BIN_NAME)?;
+
+    // `--fix` should only apply safe fixes, but should tell the user about `--fix --unsafe` if
+    // there are remaining unsafe fixes.
+    let output = cmd
+        .args([
+            "-",
+            "--format",
+            "text",
+            "--isolated",
+            "--select",
+            "F601,UP034",
+        ])
+        .write_stdin("x = {'a': 1, 'a': 1}\nprint(('foo'))\n")
+        .assert()
+        .failure();
+    assert_eq!(
+        str::from_utf8(&output.get_output().stdout)?,
+        r#"-:1:14: F601 [*] Dictionary key literal `'a'` repeated
+-:2:7: UP034 [*] Avoid extraneous parentheses
+Found 2 errors.
+[*] 1 potentially fixable with the --fix option.
+[*] 2 potentially fixable with the --fix --unsafe options.
+"#
+    );
+
+    Ok(())
+}
+
+#[test]
+fn only_unsafe_fixes_remain() -> Result<()> {
+    let mut cmd = Command::cargo_bin(BIN_NAME)?;
+
+    let output = cmd
+        .args(["-", "--format", "text", "--isolated", "--select", "F601"])
+        .write_stdin("x = {'a': 1, 'a': 1}\n")
+        .assert()
+        .failure();
+    assert_eq!(
+        str::from_utf8(&output.get_output().stdout)?,
+        r#"-:1:14: F601 [*] Dictionary key literal `'a'` repeated
+Found 1 error.
+[*] 1 potentially fixable with the --fix --unsafe options.
+"#
+    );
+
+    Ok(())
+}
+
+#[test]
+fn applies_safe_fixes_only() -> Result<()> {
```

Comment:
@MichaReiser was hoping you could validate this behavior, please. 

Do we want to show the full diagnostic summary (similar to [this](https://github.com/astral-sh/ruff/pull/5119/files#diff-92738d4e248765a1e48e3c930a03e20ed306f9b0c23dd67ad72912133dbb10f2R324)) in this case?